neutron ic querier docs

bekauz opened the pull request:
work towards #117 
builds upon #143 (this should be merged after middleware docs)

this pr introduces the initial version of the Neutron IC Querier docs.
this work is not complete - multiple sections in the pr propose ideas and potential directions rather than outline some existing outcome.

in addition to the IC Querier section, it will also bring in the storage account documentation.

uditvira commented on docs/src/libraries/neutron_ic_querier.md:
```diff
@@ -0,0 +1,264 @@
+
+This library may take on one of two possible routes with respect to how long it is considered
+active:
+1. single-use, instantiated to perform a particular query and complete
```
Single-use makes sense to me purely to stay consistent with other libraries. But single use only means that it is associated with a fixed set of output (storage) accounts. You could consider it making multiple queries that all get written to that same account.

uditvira commented on docs/src/libraries/neutron_ic_querier.md:
```diff
@@ -0,0 +1,264 @@
+For single-use design, one of the main questions is at what point does the query
+actually get registered.
+
+One approach for the single-use approach could be to register the query as soon as
```
Yes, we should allow the Library to be instantiated with a pre-configured queries.

uditvira commented on docs/src/libraries/neutron_ic_querier.md:
```diff
@@ -0,0 +1,264 @@
+written to its input account. This may involve something along the lines of:
+`if balance > 0, do x; otherwise, do y;`.
+
+It is a little less obvious as to what is the *input account* of Neutron IC Querier.
```
There is only one account associated with the Library and it's the account that the Library has permissions to write to the storage to. This breaks our previous conventions around Input/Output accounts. 

Previously, until now, Input accounts were accounts funds were take from and output accounts were accounts where funds were sent to. Or in other words, the Library had execute permissions on Input Accounts. The library did not have execute permissions around Output accounts.

In this case, we want the Library to have write permissions to the storage of an account.

uditvira commented on docs/src/libraries/neutron_ic_querier.md:
```diff
@@ -0,0 +1,264 @@
+It is a little less obvious as to what is the *input account* of Neutron IC Querier.
+For one, the input account could be standard type account that deals with token transfers.
+Prior to registering the interchain query, input account could be expected to receive
+the query deposit (in `untrn`) meant to cover the query registration costs.
```
This is not what we do in the ntrn ibc transfer library which also requires a fee. That is the closest analogous library

uditvira reviewed (commented):
Let's discuss these in our call tomorrow.

bekauz commented on docs/src/libraries/neutron_ic_querier.md:
```diff
@@ -0,0 +1,264 @@
+## Query lifecycle
+
+After `RegisterInterchainQuery` message is submitted, `interchainqueries` module will deduct
+the query registration fee from the caller.
```
I see your point here. I left it a little ambiguous on purpose because I didn't want to make assumptions on who the actual caller may be in the final design - account or library. Regardless, if it is the library, what do you think of:
> from the caller (which is this Library).

Obviously I'm far from an expert in grammar but the structure "After `RegisterInterchainQuery` message is submitted, <...>, this Library." made me a little confused on the first pass. Could be skill issue on my end.

bekauz commented on docs/src/libraries/neutron_ic_querier.md:
```diff
@@ -0,0 +1,264 @@
+
+### Query registration fee
+
+Neutron `interchainqueries` module is configured to escrow a fee (denominated in `untrn`) in order to register a query.
```
yeah true. what do you think if we introduced a markdown linter to break lines everywhere consistently? we could run it along with the documentation build job for consistency

bekauz commented:
these docs are to become slightly out of date once #158 is merged. any unaddressed comments above & updates to the new design will be done in a follow up pr.
